components/server/src/workspace/workspace-starter.ts

```diff
@@ -838,11 +838,18 @@ export class WorkspaceStarter {
             await this.tryEnableConnectionLimiting(featureFlags, user, billingTier);
             await this.tryEnablePSI(featureFlags, user, billingTier);
 
-            let usageAttributionId = await this.userService.getWorkspaceUsageAttributionId(user, workspace.projectId);
             // if the workspace has been created in an organization, we need to use the organization's attribution ID
-            if (workspace.organizationId) {
-                const org = await this.teamDB.findTeamById(workspace.organizationId);
-                usageAttributionId = AttributionId.create(org!);
+            let usageAttributionId: AttributionId | undefined = workspace.organizationId
```

Comment:
How could this actually happen (undefined)? Shouldn't every workspace have exactly one attribution ID?

Reply:
the organizationId on workspaces is currently optional until all users have been migrated to org-only mode. at that point we make it mandatory and can clean up these places (search for usages of `isMigratedToTeamOnlyAttribution`).

Reply:
Ah, that's because the Org Mode is independent of the Attribution logic so we can't always use them in a 1-1 mapping.